packages/frontend/src/fetcher/on-chain/index.ts

```diff
@@ -108,62 +112,70 @@ class Fetcher implements IPartialFetcher {
     }
 
     public async fetchAnswersHistory({
-        provider,
+        nodeClient,
         realityV3Address,
         questionId,
     }: FetchAnswersHistoryParams): Promise<RealityResponse[]> {
         if (!realityV3Address || !questionId) return [];
-        const { chainId } = await provider.getNetwork();
+        const chainId = await nodeClient.getChainId();
         enforce(
             chainId in SupportedChainId,
             `unsupported chain with id ${chainId}`
         );
-        const realityContract = new Contract(
-            realityV3Address,
-            REALITY_ETH_V3_ABI,
-            provider
-        );
 
-        let toBlock = await provider.getBlockNumber();
+        let toBlock = await nodeClient.getBlockNumber();
         let fromBlock = toBlock - LOGS_BLOCKS_SIZE;
         const answersFromLogs: RealityResponse[] = [];
         try {
             while (true) {
-                const logs = await provider.getLogs({
-                    address: realityContract.address,
+                // FIXME: check if this implementation still works properly
+                const newAnwersEventLogs = await nodeClient.getLogs({
+                    address: realityV3Address as `0x${string}`,
+                    event: parseAbiItem(
+                        "event LogNewAnswer(bytes32,bytes32,bytes32,address,uint256,uint256,bool)"
+                    ),
                     fromBlock,
                     toBlock,
-                    topics: [
-                        [NEW_ANSWER_LOG_TOPIC, NEW_QUESTION_LOG_TOPIC],
-                        questionId,
-                    ],
                 });
+                const newQuestionEventLogs = await nodeClient.getLogs({
+                    address: realityV3Address as `0x${string}`,
+                    event: parseAbiItem(
+                        "event LogNewQuestion(bytes32,address,uint256,string,bytes32,address,uint32,uint32,uint256,uint256)"
+                    ),
+                    fromBlock,
+                    toBlock,
+                });
+                newQuestionEventLogs;
+
+                const logs = [...newAnwersEventLogs, ...newQuestionEventLogs];
                 let shouldBreak = false;
                 for (const log of logs) {
+                    // FIXME: is this condition still valid
                     if (log.topics[0] === NEW_ANSWER_LOG_TOPIC) {
                         // Event params: bytes32 answer, bytes32 indexed question_id, bytes32 history_hash, address indexed user, uint256 bond, uint256 ts, bool is_commitment
-                        const [answerer] = defaultAbiCoder.decode(
-                            ["address"],
-                            log.topics[2]
+                        const [answerer] = decodeAbiParameters(
+                            [{ type: "address", name: "answerer" }],
+                            log.topics[2] as `0x${string}`
```

Comment:
Aren't `topics` typed as `Hex[]` already?

Reply:
Yes but it could have been also `undefined`, solved this way c8244b9e79bc6f23a4dfeb1d098282ee02755f81.